instrumentation/pekko/pekko-http-1.0/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/pekkohttp/v1_0/server/route/RouteConcatenationInstrumentation.java

```diff
@@ -27,6 +33,19 @@ public void transform(TypeTransformer transformer) {
         named("$anonfun$$tilde$2"), this.getClass().getName() + "$Apply2Advice");
   }
 
+  public static class OnExitFinalizer implements PartialFunction<Try<RouteResult>, Unit> {
+    @Override
+    public boolean isDefinedAt(Try<RouteResult> x) {
+      return true;
+    }
+
+    @Override
+    public Unit apply(Try<RouteResult> v1) {
+      PekkoRouteHolder.restore();
+      return null;
+    }
+  }
```

Comment:
Could you move this class to a top level class or to some other util class. We usually don't add anything besides the advice classes here.